internal/controller/predicates/predicates.go

```diff
@@ -33,6 +33,15 @@ func (GrafanaPodRecreatedPredicate) Update(e event.UpdateEvent) bool {
 	if !ok {
 		return false
 	}
+
+	if newPod == nil {
```

Comment:
This is only for update events. Deletions are not part of it but why not

Reply:
This is required otherwise unit tests fail in CI https://app.circleci.com/pipelines/github/giantswarm/observability-operator/1671/workflows/93d023d3-8b41-4bdc-8be9-5109d659cce4/jobs/7242?invite=true#step-102-18005_67